forem/forem
# Twitter OAuth2

## What type of PR is this? (check all applicable)


## Description

## Related Tickets & Documents

- Closes #18192

## QA Instructions, Screenshots, Recordings

1. Boot app locally
2. Enabled `:twitter_oauth2` FeatureFlag
3. Go to developers.twitter.com and create an app that allows OAuth2 login ([more details on what this looks like here](https://github.com/forem/forem/issues/18192#issuecomment-1353670887))
4. Fill in Client ID and Client Secret on the admin dashboard on `/admin/customization/config`
5. You should be able to sign in/up with Twitter and also connect an existing account via `/settings`

### UI accessibility concerns?

None - the UI remains the same

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: We are releasing this new Twitter OAuth option behind a Feature Flag to test this on a production environment and once confirmed we are good to remove the Feature Flag we will communicate the change appropriately

## [optional] Are there any post deployment tasks we need to perform?

Test this on Canary/CodeNewbie community

## [optional] What gif best describes this PR or how it makes you feel?

![upgrade - lemonade stand](https://media.giphy.com/media/LnoY34a2mcGm3J5XEw/giphy.gif)
